dimagi/commcare-hq
# Make locations system field

## Product Description

Should have no visible changes, except for user cannot set `commcare_location_id`, `commcare_location_ids`, and `commcare_primary_case_sharing_id` in UserData directly 

## Technical Summary

Ticket: https://dimagi.atlassian.net/browse/SAAS-15716

Please review by commit.


## Safety Assurance

### Safety story

Write new tests for it.
This PR only changes where we pull data from, UserData or the `location_id` and `assigned_location_ids` attribute on CouchUser.

### Automated test coverage


`corehq.apps.user_importer.tests.test_importer:TestMobileUserBulkUpload` ensures location in user data will be updated when use bulk upload to update user's location.
New tests added in `corehq.apps.users.tests.test_user_data:TestUserDataModel` ensure location in user data will be updated when user's location changed.
### QA Plan


QA Ticket: https://dimagi.atlassian.net/browse/QA-6781

QA plan:
- make changes to a mobile worker's location, and use User API to list all mobile workers, make sure what is returned in user data is expected. (If no location assigned, then no `location_id`, `location_ids` in user data)
- try to update `location_id` and `location_ids` in user data through User API and get error.



### Rollback instructions




### Labels & Review
